File: MoroTest/src/main/java/com/moro/MoroTest/controller/CurrentUser.java
package com.moro.MoroTest.controller;

import org.springframework.security.core.annotation.AuthenticationPrincipal;

import java.lang.annotation.*;

//https://docs.spring.io/spring-security/site/docs/4.0.0.RELEASE/reference/html/mvc.html
@Target({ElementType.PARAMETER, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@AuthenticationPrincipal
public @interface CurrentUser {}

File: MoroTestKotlin/src/main/kotlin/com/moro/MoroTestKotlin/annotation/CurrentUser.kt
package com.moro.MoroTestKotlin.annotation

import org.springframework.security.core.annotation.AuthenticationPrincipal

//https://docs.spring.io/spring-security/site/docs/4.0.0.RELEASE/reference/html/mvc.html
@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@AuthenticationPrincipal
annotation class CurrentUser